### PR TITLE
Cleanup font subsetting code

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

I noticed a few things while reviewing font-related PRs, but as they were not directly related to those PRs, did not comment on them (except for the Type 42 PS subsetting bit, as that one was merged before I could comment.)

See commit messages for details, but this removes redundant character<->ordinal conversion, splits Type 42 PS subsetting into a separate function, and removes some deprecated code.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/backends/backend_pdf.py [hunk 1]
```diff
@@ -1227,13 +1227,9 @@ def embedTTFType42(font, characters, descriptor):
...
 
-            _log.debug(
-                "SUBSET %s characters: %s",
-                filename, "".join(chr(c) for c in characters)
-            )
-            fontdata = _backend_pdf_ps.get_glyphs_subset(
-                filename, "".join(chr(c) for c in characters)
-            )
+            subset_str = "".join(chr(c) for c in characters)
+            _log.debug("SUBSET %s characters: %s", filename, subset_str)
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pdf.py:1231]: Optional: f-string

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/backends/backend_pdf.py (hunk 1 above)
Comment (QuLogic) [reply lib/matplotlib/backends/backend_pdf.py:1231]: Logging doesn't use f-strings, usually.

**Review by jkseppan — APPROVED** [review]

**Comment by jkseppan** [comment]
Looks good to me, I wonder if @sauerburger wants to take a look? If not, let's just merge.

**Review by sauerburger — APPROVED** [review]
Looks good to me

**Pull request merged**